test/dynamo/test_graph_break_messages.py

```diff
@@ -0,0 +1,221 @@
+            """\
+Dynamic shape operator
+  Explanation: Operator `aten.nonzero.default`'s output shape depends on input Tensor data.
+  Hint: Enable tracing of dynamic shape operators with `torch._dynamo.config.capture_dynamic_output_shape_ops = True`
+
+  Developer debug context: aten.nonzero.default
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    return torch.nonzero(torch.rand([10, 10]))""",
+        )
+
+    def test_dynamic_shape_operator_no_meta_kernel(self):
+        def fn():
+            return torch.bincount(torch.randint(0, 10, (10,)))
+
+        with torch._dynamo.config.patch(capture_dynamic_output_shape_ops=True):
+            self.assertExpectedInlineMunged(
+                Unsupported,
+                lambda: torch.compile(fn, backend="eager", fullgraph=True)(),
+                """\
+Dynamic shape operator (no meta kernel)
+  Explanation: Operator `aten.bincount.default` does not have a meta kernel that supports dynamic output shapes
+  Hint: Please report an issue to PyTorch
+
+  Developer debug context: aten.bincount.default
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    return torch.bincount(torch.randint(0, 10, (10,)))""",
+            )
+
+    def test_data_dependent_operator(self):
+        def fn(x):
+            return x.item()
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(torch.Tensor([1])),
+            """\
+Tensor.item
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    return x.item()""",
+        )
+
+    def test_data_dependent_operator2(self):
+        def fn(x):
+            return torch.equal(x, x)
+
+        with torch._dynamo.config.patch(capture_scalar_outputs=True):
+            self.assertExpectedInlineMunged(
+                Unsupported,
+                lambda: torch.compile(fn, backend="eager", fullgraph=True)(torch.ones(3)),
+                """\
+Data dependent operator
+  Explanation: Operator `aten.equal.default` has a non-Tensor output whose value is dependent on the data of Tensor inputs.
+  Hint: Consider wrapping the operator into a PyTorch-understood custom operator (see https:/pytorch.org/tutorials/advanced/custom_ops_landing_page.html)
+
+  Developer debug context: aten.equal.default
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    _helper(fn, (torch.ones(3),)),""",
+            )
+
+    def test_super_call_method(self):
+        def fn(it):
+            return [x for x in it]
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(zip(range(5), range(10))),
+            """\
+Unsupported method call
+  Explanation: Dynamo does not know how to trace method `__iter__` of class `zip`
+  Hint: Avoid calling `zip.__iter__` in your code.
+  Hint: Please report an issue to PyTorch.
```

Comment:
Richard: We do support zip, we just don't support passing zip as an input to the torch.compile'd region. 

Ryan: This might need to be a longer explanation. The reason why the user has run into this is because they have a graph break somewhere else that is triggering this, they should go looking for the actual graph break that triggered it.

Animesh: This may be a fundamental graph break. We could skip the frame even before we start tracing.